src/foam/nanos/medusa/MedusaAdapterDAO.js

```diff
@@ -92,13 +111,55 @@ It then marshalls it to the primary mediator, and waits on a response.`,
       return formatter;
     }
   };
+
+  // TODO: pool of reusable WriteLocks
+  protected Map<Object, WriteLock> findLocks_ = new HashMap();
```

Comment:
How is this collection synchronized?

Reply:
yes, multiple puts of the same id need synchronization.